app/assets/javascripts/credit_card_form.js

```diff
@@ -0,0 +1,63 @@
+$(document).on("ready turbolinks:load", function () {
+  let show_error, stripeResponseHandler, submitHandler;
+
+  submitHandler = function (event) {
+    const $form = $(event.target);
+    $form.find("input[type=submit]").prop("disabled", true);
+
+    //If Stripe was initialized correctly this will create a token using the credit card info
+    if (Stripe) {
+      Stripe.card.createToken($form, stripeResponseHandler);
+    } else {
+      show_error(
+        "Failed to load credit card processing functionality. Please reload this page in your browser."
+      );
+    }
```

Comment:
You should do this before line 6.

Reasons:
1. Don't have to hit the DOM to pull out the node
2. If `Stripe` is undefined, your submit input will be disabled. To reenabled it, you'll have to refresh the page <= I didn't test but it should be like this based on the code unless I am missing something

Reply:
Hmm I don't quite see why the `show_error` block should be ran before line 6. Wouldn't line 6 still run regardless of whether Stripe is defined or not in this particular case since the `show_error` will just show the flash message if it is undefined?

Reply:
I meant, instead of:
```
const $form = $(event.target);
$form.find("input[type=submit]").prop("disabled", true);

//If Stripe was initialized correctly this will create a token using the credit card info
if (Stripe) {
  Stripe.card.createToken($form, stripeResponseHandler);
} else {
  show_error(
    "Failed to load credit card processing functionality. Please reload this page in your browser."
  );
}
```
do
```
if(!Stripe) {
  show_error(
    "Failed to load credit card processing functionality. Please reload this page in your browser."
  );
  return
}
const $form = $(event.target);
$form.find("input[type=submit]").prop("disabled", true);

//If Stripe was initialized correctly this will create a token using the credit card info
Stripe.card.createToken($form, stripeResponseHandler);
```
Doing it like this means your code won't hit the dom to find the input element. Also, it would need to initialise `$form` if `Stripe` is undefined / null